### DISCUSSION TITLE
Better Self-Hosted Install Documentation (Without Using Docker)

### DISCUSSION THREAD

**Discussion by stevefloyd** [uxbox/uxbox#751] [Ideas]
In reading through the existing documentation on how to deploy this app on a server, it basically says "there are other ways to install it, but just use Docker". 

It would be nice if someone from the project or community could put together some documentation on how to deploy to your own server WITHOUT using Docker.

Maybe there is something like this already, I just haven't seen it. If anyone knows please post any resources you have for this approach.

**Comment by niwinz** [comment]
Hi @stevefloyd,

You right, we have no installation guide outside of docker. We have it in our roadmap. 

As you may know, we released the penpot to the public 1 month ago aprox as first alpha, and we prefer have a simple but single install guide than no install guide at all. For now we have the docker based one... In future we will work in alternatives. 

On the other hand, I think the title is a little bit missleading, because the current guide is good enough if you are ok on using docker, so you **don't want a better guide**, you want a **different/alternative installation guide**.

Do you have any suggestion that we can incorporate to our roadmap about which installation guide you want/need?

**Reply by stevefloyd** [reply]
Someone just needs to create a [Yunohost](https://yunohost.org/en/packaging_apps_start) package. 

That would probably unlock a ton of users and testing in various environments. That’s your core audience anyways (creatives & devs).

Any OTP Linux installation documentation that doesn’t rely on Docker would be good though.

I may just do it myself.

**Reply by hirunatan** [reply]
We have been thinking about this. At some point we'll write an Ansible role with the detailed steps to build a Penpot instance. This will be more explicit, testable and maintainable than writing a textual documentation, and would serve as a basis for building linux distribution packages, or apps for Yunohost, Sandstorm or Cloudron, for example.

We have it in our roadmap, although perhaps we'll wait a bit more, because in this Alpha stage we are still making big changes on the deployment setup, and it's complicated to maintain if we have many packaging and distribution formats.

**Comment by VladimirLevadnij** [comment]
@niwinz Hello! I want to know if there is an instruction to install / run penpot without using docker?

I want to make changes to the code in a simple way for testing purposes. That is, I want to modify the code locally and run the result. I know that it is possible to do this somehow with docker, but so far I do not know how to do this and it seems to me that this is a difficult way. Or I'm wrong?

Also, I would like to be able to run all 5 parts of Penpot, which are now running as 5 docker containers, on separate cloud servers that run the Linux Ubuntu operating system without using docker. Can I read somewhere how to do this?

**Reply by SoSoWow** [reply]
How do you plan to have the proficiency to "modify the code" and  "make changes to the code for testing purposes", but you don't know how to simply pull code from Git and/or compile and install the code? 
I'm running into the same problem when I'm trying to rewrite this Physicist dude Leonard Susskind's papers on String Theory, you know kinda put my own little "particle spin" on it (🤣), then test my remix and release my version called, "Super Sick String Theory", but these hatin ass, nerdy wanna be "physicist" are coppin this huge attitude cause I asked them to help me with the Algebra so I could tweak the theory, cause Im not so good at maths... But noooo, now they playing like they too busy to even respond to me, like his schedule too busy as a Cosmologist at Stanford or something.... People got attitude ya know? So I feel ya on that braddha, rEspEKt!

**Reply by stevefloyd** [reply]
Be nice, this repo isn’t the place for mocking people.

**Reply by VladimirLevadnij** [reply]
Hello SoSoWow. I am a normal project manager. I set tasks and drive out people who do not know how to properly allocate their time. For example, such people who, instead of writing instructions on the case, become personal, do not know how to build social communications for effective teamwork :-)

As for my question that I asked above, as I have a lot of experience with various projects and my responsibility as a project manager, in order for the result to be successful, is not to assume something, not to invent what something, but in finding clear and concise instructions, including those presented in the form of flowcharts. I am sure that one day you too will be able to become a specialist who understands such things. I wish you success in your work and leisure! :-)

**Comment by Verequies** [comment]
For those here that would like to install penpot without docker, I have uploaded my build and run scripts as apart of issue  https://github.com/penpot/penpot/issues/1917 . Everything seems to work great so far except for uploading images as per the issue.

**Reply by airtonix** [reply]
Without looking at the details...

Isn't the dockerfile itself the documentation?

Or have the penpot team abstracted the creation of the docker images?

**Reply by niwinz** [reply]
dockerfiles, and build scripts are all available on the repository

**Comment by andar1an** [comment]
I think it is pretty great to have a non-containerized  or virtualized option.

There are some great examples of self-hosted services installed via linux distro package managers that use various init systems e.g. openrc or systemd. 

Between those references and the dockerfiles available, it shouldn't be too bad.

Jellyfin is usually a good example as it has a web client and databases too if anyone needs an additional reference.
